### ISSUE TITLE
Premise selection/Proof step classification?

### ISSUE THREAD

**Issue by spolu** [princeton-vl/FormulaNet#1]
From reading [0] it seems that you compare your approach to HolStep's baselines which operate on the task of "proof step classification" (not premise selection) as stated in [1]. Wouldn't premise selection more precisely consists in selecting the dependency formulas (prefixed by `D` in the dataset) rather than classifying the useful proof steps (prefixed by `+`/`-`)? What is the rationale for titling [0] premise selection instead of proof step classification?

Thanks!

[0] https://arxiv.org/pdf/1709.09994.pdf
[1] https://arxiv.org/pdf/1703.00426.pdf

**Comment by mingzhew** [comment]
@spolu Yes, our experiment on HolStep are about classifying proof steps. Actually premise selection and proof step classification share the same specification that take a conjecture and a statement as input and output true or false. So our method can be used on both of them and we refer such problem as premise selection. But you are right, if you really want to distinguish classification on dependency formulas and proof steps, our experiment is about proof steps.

**Comment by spolu** [comment]
Well premise selection has the increased complexity that the input statement set is potentially very large (depending on the setup, but extremely large for general premise selection). Have you given any thoughts as to how to tackle that extra complexity?

**Comment by BJ-peker** [comment]
Hi, when I ran the batch_test.py, the command is " python batch_test.py --model FormulaNet-basic  --log log_test.txt --dict_file data/dicts/hol_train_dict",but an error occured .
The infomation is "	size mismatch for l1.weight: copying a param with shape torch.Size([256, 1909]) from checkpoint, the shape in current model is torch.Size([1909, 256])."  
I am not major in pytorch ,so it will be very perfect if you can help me. Thanks!
